packages/store/operators/src/remove-item.ts

```diff
@@ -0,0 +1,33 @@
+type Predicate<T = unknown> = (value?: T) => boolean;
+
+/**
+ * @param selector - index or predicate to remove an item from an array by
+ */
+export function removeItem<T>(selector: number | Predicate<T>) {
+  return function removeItemOperator(existing: Readonly<T[]>): T[] {
+    let index = -1;
+
+    if (isPredicate(selector)) {
+      index = existing.findIndex(selector);
+    } else if (isNumber(selector)) {
+      index = selector;
+    }
+
+    // If non-existing index was provided
+    if (isNaN(index) || index === -1 || !existing.hasOwnProperty(index)) {
+      return existing;
+    }
+
+    const clone = [...existing];
+    clone.splice(index, 1);
```

Comment:
usage `filter` instead `copy` and `splice` to make it faster

before

```ts
const clone = [...existing];
clone.splice(index, 1);
```

after

```ts
existing.filter((_, i) => i !== index); 
```

Reply:
clone = [...existing.slice(0, index), ...existing.slice(index + 1)];

Fast

Reply:
@splincode have to make benchmarks, will answer later

Reply:
@splincode are you still there? :smile: 

![image](https://user-images.githubusercontent.com/7337691/51426983-729a4e80-1bfa-11e9-941e-689f9fe637fc.png)